tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,156 @@
+from unittest.mock import ANY, patch
+
+import numpy as np
+from harmony.util import config
+from netCDF4 import Dataset
+from numpy.testing import assert_array_equal
+from varinfo import VarInfoFromDmr
+
+from hoss.coordinate_utilities import (
+    get_coordinate_variables,
+    get_fill_values_for_coordinates,
+    get_geo_grid_corners,
+    get_lat_lon_arrays,
+    get_override_projected_dimension_name,
+    get_override_projected_dimensions,
+    get_row_col_sizes_from_coordinate_datasets,
+    get_valid_indices,
+    get_variables_with_anonymous_dims,
+    get_x_y_extents_from_geographic_points,
+    is_lat_lon_ascending,
+    update_dimension_variables,
+)
+from hoss.exceptions import MissingCoordinateDataset, MissingValidCoordinateDataset
+
+
+class TestCoordinateUtilities(TestCase):
+    """A class for testing functions in the `hoss.coordinate_utilities`
+    module.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create fixtures that can be reused for all tests."""
+        cls.config = config(validate=False)
+        cls.logger = getLogger('tests')
+        cls.varinfo = VarInfoFromDmr(
+            'tests/data/SC_SPL3SMP_008.dmr',
+            'SPL3SMP',
+            config_file='hoss/hoss_config.json',
+        )
+        cls.nc4file = 'tests/data/SC_SPL3SMP_008_prefetch.nc4'
+        cls.latitude = '/Soil_Moisture_Retrieval_Data_AM/latitude'
+        cls.longitude = '/Soil_Moisture_Retrieval_Data_AM/longitude'
+
+        cls.lon_arr = np.array(
+            [
+                [-179.3, -120.2, -60.6, -9999, -9999, -9999, 80.2, 120.6, 150.5, 178.4],
+                [-179.3, -120.2, -60.6, -9999, -9999, -9999, 80.2, 120.6, 150.5, 178.4],
+                [-179.3, -120.2, -60.6, -9999, -9999, -9999, 80.2, 120.6, 150.5, 178.4],
+                [-179.3, -120.2, -60.6, -9999, -9999, -9999, 80.2, 120.6, 150.5, 178.4],
+                [-179.3, -120.2, -60.6, -9999, -9999, -9999, 80.2, 120.6, 150.5, 178.4],
+            ]
+        )
+
+        cls.lat_arr = np.array(
+            [
+                [89.3, 89.3, -9999, 89.3, 89.3, 89.3, -9999, 89.3, 89.3, 89.3],
+                [50.3, 50.3, 50.3, 50.3, 50.3, 50.3, -9999, 50.3, 50.3, 50.3],
+                [1.3, 1.3, 1.3, 1.3, 1.3, 1.3, -9999, -9999, 1.3, 1.3],
+                [-9999, -60.2, -60.2, -9999, -9999, -9999, -60.2, -60.2, -60.2, -60.2],
+                [-88.1, -88.1, -88.1, -9999, -9999, -9999, -88.1, -88.1, -88.1, -88.1],
+            ]
+        )
+
+    def setUp(self):
+        """Create fixtures that should be unique per test."""
+        self.temp_dir = mkdtemp()
+
+    def tearDown(self):
+        """Remove per-test fixtures."""
+        if exists(self.temp_dir):
+            rmtree(self.temp_dir)
+
+    def test_get_valid_indices(self):
+        """Ensure that latitude and longitude values are correctly identified as
+        ascending or descending.
+
+        """
+
+        expected_result1 = np.array([0, 1, 2, 3, 4])
```

Comment:
Something of a theme here - there are a bunch of variable names in these tests that are really vague (`expected_result1`, `expected`, `expected_result`, similar things with `actual`). It would be clearer to the reader if the names were more specific. Something like `expected_valid_indices`.

Unit tests are a secondary form of documentation for developers, and can be really informative. It's really helpful (to me at least 😅) if they are as easy to read as possible.

Reply:
done - de350b62112b7fef73f2184a60196c25368ccd12